prawcore/auth.py

```diff
@@ -367,18 +401,28 @@ def __init__(
         :param two_factor_callback: A function that returns OTPs (One-Time
             Passcodes), also known as 2FA auth codes. If this function is
             provided, prawcore will call it when authenticating.
+        :param scopes: (Optional) A list of OAuth scopes to request authorization for
+            (default: None). The scope ``*`` is requested when the default argument is
+            used.
 
         """
         super(ScriptAuthorizer, self).__init__(authenticator)
         self._username = username
         self._password = password
         self._two_factor_callback = two_factor_callback
+        self._scopes = scopes
 
     def refresh(self):
         """Obtain a new personal-use script type access token."""
+        additional_kwargs = {}
+        if self._scopes:
+            additional_kwargs["scope"] = " ".join(self._scopes)
+        otp = self._two_factor_callback and self._two_factor_callback()
+        if otp:
+            additional_kwargs["otp"] = otp
```

Comment:
While Reddit uses the parameter name `otp` (1) that's not what this actually is, and (2) that variable name doesn't adhere to the standards of praw/prawcore as it may not be clear to someone reading the code.

```suggestion
        two_factor_code = self._two_factor_callback and self._two_factor_callback()
        if two_factor_code:
            additional_kwargs["otp"] = two_factor_code
```